private/lib/Controller/Template.php

```diff
@@ -0,0 +1,174 @@
+
+        $templateId = $this->service->createTemplate($this->getUserId(), $categoryId, $templateTitle, $templateContent);
+
+        $this->setNotification(Notification::TYPE_SUCCESS, "Template {$templateTitle} has been created");
+
+
+        Redirect::to(self::TEMPLATES_URL);
+    }
+
+    public function createView(): void
+    {
+        $this->injectSessionVariableToTemplate();
+        $categories = $this->categoryService->getAllCategoriesForUser($this->getUserId());
+
+        $this->template->setVariable('categories', $categories);
+        $this->template->render('template/create');
+    }
+
+    public function update(): void
+    {
+        $this->template->setVariable('post', $_POST);
+
+        /** @see TemplateValidator::update() */
+        $this->validator->validate(__FUNCTION__);
+
+        $categoryId      = Sanitize::int($_POST['category_id']);
+        $templateId      = Sanitize::int($this->getRouteParameters()['id']);
+        $templateTitle   = Sanitize::string($_POST['template_title'], 'strip|capitalize');
+        $templateContent = Sanitize::string($_POST['template_content'], 'trim');
+
+        $this->service->updateTemplate(
+            $this->getUserId(),
+            $categoryId,
+            $templateId,
+            $templateTitle,
+            $templateContent
+        );
+
+        $this->setNotification(
+            Notification::TYPE_SUCCESS,
+            "Template {$templateTitle} has been updated"
+        );
+
+        Redirect::to(self::TEMPLATES_URL);
+    }
+
+    public function updateView(): void
+    {
+        $this->injectSessionVariableToTemplate();
+
+        $templateId = Sanitize::int($this->getRouteParameters()['id']);
+
+        try {
+            $categories = $this->categoryService->getAllCategoriesForUser($this->getUserId());
+            $template = $this->service->getTemplateForUser($templateId, $this->getUserId());
+
+            $this->template->setVariables([
+                'template' => $template,
+                'categories' => $categories,
+            ]);
+        } catch (UserException $e) {
+            $this->template->setVariable(
+                Notification::TYPE_ERROR,
+                $e->getMessage()
+            );
+        }
+
+        $this->template->render('template/update');
+    }
+
+    public function delete(): void
+    {
+        $templateId = Sanitize::int($this->getRouteParameters()['id']);
+
+        // setting get variable for validator
+        $_GET['form_key'] = $this->getRouteParameters()['antiCsrfToken'];
+
+        /** @see TemplateValidator::delete() */
+        $this->validator->validate(__FUNCTION__);
+
+        $this->service->deleteTemplate($templateId, $this->getUserId());
+
+        $this->setNotification(Notification::TYPE_SUCCESS, 'Template was removed');
+
+        $this->deleteView();
+    }
+
+    public function deleteView(): void
+    {
+        $this->injectSessionVariableToTemplate();
```

Comment:
Since you're not actually loading a template page, there is no need to have this here. Please remove it